pages/_app.tsx

```diff
@@ -1,12 +1,38 @@
 import Head from 'next/head';
-
 import { ChakraProvider } from '@chakra-ui/react';
+import { noSSR } from 'next/dynamic';
+import React, { useEffect } from 'react';
 
-import React from 'react';
+import { setTelemetry, useTelemetry } from '../hooks/useTelemetry';
 
 import './style.css';
+import { BrowserMetricsProvider } from '../lib/types';
 
 function App({ Component, pageProps }) {
+  const telemetry: BrowserMetricsProvider|null = useTelemetry()
+  /**
+   * We have to do funky imports here to satisfy nextjs since this package is cjs and ignite-metrics is esm
+   */
+  useEffect(() => {
+    (async() => {
+      // @ts-expect-error
+      const { BrowserMetricsProvider } = await noSSR(() => import('@ipfs-shipyard/ignite-metrics/browser-vanilla'), {})
+
+      setTelemetry(new BrowserMetricsProvider({ appKey: '294089175b8268e44bc4e4fab572fe250d57b968' }))
```

Comment:
nit: do we want this key exposed?

Reply:
yea it's fine. we could hide it on the backend but we can only do that on this project, the rest will have the key in the source code